app/managers/security_manager/index.ts

```diff
@@ -0,0 +1,477 @@
+        }
+
+        if (activeServer && (!added.has(activeServer) || !this.activeServer)) {
+            this.activeServer = activeServer;
+            this.setScreenCapturePolicy(activeServer);
+            const isJailbroken = await this.isDeviceJailbroken(activeServer);
+            if (!isJailbroken) {
+                this.authenticateWithBiometricsIfNeeded(activeServer);
+            }
+        }
+    }
+
+    /**
+     * Handles app state changes to prompt authentication when resuming from background.
+     */
+    onAppStateChange = async (appState: AppStateStatus) => {
+        if (this.isAuthenticationHandledByEmm()) {
+            return;
+        }
+
+        const isActive = appState === 'active';
+        const isBackground = appState === 'background';
+
+        if (isActive && this.previousAppState === 'background') {
+            if (this.activeServer) {
+                const config = this.getServerConfig(this.activeServer);
+                if (config && config.Biometrics && isMainActivity()) {
+                    const authExpired = this.backgroundSince > 0 && (Date.now() - this.backgroundSince) >= toMilliseconds({minutes: 5});
+                    if (authExpired) {
+                        const isJailbroken = await this.isDeviceJailbroken(this.activeServer);
+                        if (!isJailbroken) {
+                            await this.authenticateWithBiometrics(this.activeServer);
+                        }
+                    }
+                    this.backgroundSince = 0;
+                }
+            }
+        } else if (isBackground) {
+            this.backgroundSince = Date.now();
+        }
+
+        this.previousAppState = appState;
+    };
+
+    /**
+     * Checks if EMM is already enabled and setup
+     * to handle biometric / passcode authentication.
+     */
+    isAuthenticationHandledByEmm = () => {
+        return ManagedApp.enabled && ManagedApp.inAppPinCode;
+    };
+
+    /**
+     * Checks if EMM is already enabled and setup
+     * to handle jailbreak protection.
+     */
+    isJalbreakProtectionHandledByEmm = () => {
+        return ManagedApp.enabled && ManagedApp.cacheConfig?.jailbreakProtection === 'true';
+    };
+
+    /**
+     * Checks if EMM is already enabled and setup
+     * to handle screenshot protection.
+     */
+    isScreenshotProtectionHandledByEmm = () => {
+        return ManagedApp.enabled && ManagedApp.cacheConfig?.blurApplicationScreen === 'true';
+    };
+
+    /**
+     * Get the configuration of a server to prevent screenshots.
+     */
+    isScreenCapturePrevented = (server: string) => {
+        const config = this.getServerConfig(server);
+        if (!config) {
+            return false;
+        }
+
+        return config.PreventScreenCapture == null ? false : config.PreventScreenCapture;
+    };
+
+    /**
+     * Checks if the device is Jailbroken or Rooted.
+     */
+    isDeviceJailbroken = async (server: string, siteName?: string) => {
```

Comment:
All the usage of `isDeviceJailbroken` in this PR that I can find do not pass siteName. So, whats the purpose? Can you add docs explaining it and how it affects the result of this function?

Reply:
sorry, I'm fixing this.. this should have `siteName` set when connecting to a new server that has this setting enabled.

The reason to use siteName is to be able to go through the check without having to add the server security config into memory if you are unable to conform with the security policies before connecting. and the Site name is being used in the Alert box that is displayed in case of failure.